### ISSUE TITLE
speedglm warnings

### ISSUE THREAD

**Issue by benkeser** [benkeser/survtmle#23]
With the new pull request merged, the vignette compiles with a bunch of warnings about `speedglm` breaking. Two questions, @nhejazi, 
1. Is that just `speedglm` breaking internally or are we making a mistake in how it's called that breaks it?
2. Would it be better to suppress those warnings? I'm worried that the warnings will give people the impression that something is wrong.

**Comment by nhejazi** [comment]
Are the warnings of the form `'speedglm' ran into an error in [calling_fun]. 'glm' will be used instead`, or something else entirely?

**Comment by benkeser** [comment]
See [the vignette](https://benkeser.github.io/survtmle/articles/survtmle_intro.html) examples.

**Comment by nhejazi** [comment]
Got it -- I think I have an idea as to the cause of this.

**Comment by nhejazi** [comment]
The `pkgdown` site was built using an old version (v.1.0.1.2) of the package. Using the current `master` branch (v.1.0.2.0) fixes the `speedglm` errors. 71f10a6 fixes this -- feel free to close this issue.

That said, a single error does persist:
```
## Warning in chol.default(B, pivot = TRUE): the matrix is either rank-
## deficient or indefinite
```
...but this seems more related to the structure of the first example in the vignette than the `speedglm` implementation.

**Comment by nhejazi** [comment]
Scratch that -- there's something wrong with how `speedglm` interacts with `SuperLearner`. This will take more digging.

**Comment by nhejazi** [comment]
Notes from preliminary investigation:
* case 1: `method = "mean"`: error in `speedglm` arises in `fluctuateIteratedMean` (to create the object `wideDataList`) and is displayed 6 times due to loop over `seq_len(nrow(timeAndType))`, starting in line 247 of the function `mean_tmle`

The above is the only case which causes the observed error.

@benkeser -- I would propose a few solutions:
(1) We avoid calling `glm_fast` in `fluctuateIteratedMean` and merely use `glm` instead.
(2) We change the output produced by `fast_glm` from a `message` to a `warning` in the case that `speedglm` fails. This will have the obvious effect of causing warnings to appear when `speedglm` does fail but mitigates this problem since `fluctuateIteratedMean` already wraps `glm_fast` in a `suppressWarnings` call (which doesn't suppress messages). We could, of course, also just add a `supressMessages` call to `fluctuateIteratedMean`.

**Comment by benkeser** [comment]
`fluctuateIteratedMean` could always call `glm` and never `speedglm`. Alternatively, would suggest checking that this isn't a `sparseMatrix` thing. The glm that we're fitting in `fluctuateIteratedMean` will only have a few covariates (maybe just one? I don't remember how I'm doing the fluctuation for multiple treatment levels) that won't be sparse.

**Comment by nhejazi** [comment]
The error that causes the `speedglm` call to fail is the following: `Error in sqrt(w) * X : non-conformable arrays`.

Also, the failure is not related to `sparseMatrix` -- setting `sparse = FALSE` in `speedglm` causes the same error above.

And here's the `traceback()`:
<img width="497" alt="screen shot 2017-08-24 at 1 26 04 pm" src="https://user-images.githubusercontent.com/8527276/29687015-dae34010-88cf-11e7-9d32-9f8dde265317.png">

**Comment by nhejazi** [comment]
SO posts about similar errors indicate that we're doing some bad linear algebra (dimensions of objects being operated upon not matching up correctly). Perhaps we're being saved by some kind of safety mechanisms built into `glm` that are absent from `speedglm`?

Personally, I'd like to fix this by simply substituting `fast_glm` for `glm` in `fluctuateIteratedMean`, since it's just a single use of GLMs and the `tryCatch` is going to cost time, but, as given above, we can also just change the output message to a warning and avoid the undesirable output.

**Comment by benkeser** [comment]
Is `speedglm` upset about the lack of intercept in the fluctuation model fit?

**Comment by benkeser** [comment]
Can you debug `speedglm.wfit` and confirm that the `intercept` option is being passed in as `FALSE`?

**Comment by nhejazi** [comment]
Sure -- I'll look into it and report back.

I'm also just testing the solution of switching `fast_glm` out for `glm` in `fluctuateIteratedMean`.

**Comment by benkeser** [comment]
I guess I have no issue with just using `glm`. That's definitely the easiest fix and, really, I doubt we're saving all that much time with `speedglm` in this particular function.

**Comment by nhejazi** [comment]
Ok, I'm just going to implement the `speedglm` to `glm` fix and save a bit of time then. I definitely agree that we're not saving computational time by using `speedglm` here.

**Comment by nhejazi** [comment]
Closing this since everything appears fixed by e3ee297.

**Comment by benkeser** [comment]
Thanks, Nima!
